### ISSUE TITLE
Accessibility box shown on click on Google Chrome

### ISSUE THREAD

**Issue by FloChehab** [Leaflet/Leaflet#8659]
### Checklist

- [X] I've looked at the [documentation](https://leafletjs.com/reference.html) to make sure the behavior isn't documented and expected.
- [X] I'm sure this is an issue with Leaflet, not with my app or other dependencies (Angular, Cordova, React, etc.).
- [X] I've searched through the [current issues](https://github.com/Leaflet/Leaflet/issues) to make sure this hasn't been reported yet.
- [X] I agree to follow the [Code of Conduct](https://github.com/Leaflet/Leaflet/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.

### Steps to reproduce

Followup to https://github.com/Leaflet/Leaflet/pull/8247

1. Go to https://plnkr.co/edit/VQ549soZtte7FRFT?preview (on Google Chrome, this issue doesn't occur on Firefox)
2. Click on the polygon with a Tooltip
3. The accessibility box appears as in the screenshot.

![image](https://user-images.githubusercontent.com/20242665/201962304-db67d123-97da-4158-9615-905230c87c5b.png)

This is particulary inadequate when custom style is applied on click such as this : 
![Screenshot from 2022-11-15 15-50-53](https://user-images.githubusercontent.com/20242665/201963499-3ea1ef80-24fb-4fd1-abc0-36121cc93a8d.png)

### Expected behavior

No accessibility box is show (or it can be disabled by a flag).

### Current behavior

An accessibility box is show on click, whether the user wants it or not, and without any configuration from the Leaflet user.

### Minimal example reproducing the issue

https://plnkr.co/edit/VQ549soZtte7FRFT?preview

### Environment

- Leaflet version: 1.9.2
- Browser (with version): Google Chrome (107.0.5304.87) (this issue doesn't occur on Firefox)
- OS/Platform (with version): tested on Fedora 26 & latest Mac OSX

**Comment by FloChehab** [comment]
Hello @Malvoz,

Here is the issue, a bit quicker than expected, I thought it would be harder to have a reproducible setup.

**Comment by chrisbitmead** [comment]
Users of our app  ( http://anbg.gov.au/fap ) are suddenly reporting the same thing, and I agree it occurs on Chrome but not Firefox. It also occurs in Edge, I don't know if that's because Edge shares the Chrome rendering engine.

**Comment by Falke-Design** [comment]
Looks like Chrome made it possible to make vectors focusable (useing tab) if a `foucs` or `blur` listener is added to the svg. A Leaflet internal attempt is #7822 and #8251.

The outline is caused by `_addFocusListenersOnLayer` in `Tooltip.js`. It adds `focus` and a `blur` listener, so Chrome shows the outline over the `:focus` style.

@Malvoz do you know something about this and do you have any idea how to prevent this if a user clicks on it?

**Comment by Malvoz** [comment]
> This is particulary inadequate when custom style is applied

@FloChehab That depends on the user, the outline is essential for some people, e.g. a color change alone is not an obvious focus style to those with color blindness or low vision. You may disable it for all of your users, but please consider [the accessibility concerns](https://developer.mozilla.org/en-US/docs/Web/CSS/outline#accessibility_concerns:~:text=a%20single%20outline.-,Accessibility%20concerns,-Assigning%20outline%20a).

Did you try the solution mentioned in https://github.com/Leaflet/Leaflet/pull/8247#issuecomment-1265897939?

> ```css
> .leaflet-container :focus:not(:focus-visible) {
>   outline: 0;
> }
> ```

As @Falke-Design stated, Chrome uses different heuristics for when to show the focus outline, personally I'd love to see other browsers follow in this matter. 🙂

If there's a consensus to add the above CSS to leaflet.css, we should consider that.

**Comment by FloChehab** [comment]
@Malvoz Thanks for coming back to this.

* Of course accessibility is important, though building some custom dataviz should be possible without "undesired" black box appearing (and have the ability to toggle the accessibility mode).
* We have just switched to Maplibregljs today but it seems like the fix you are suggesting is working on the reproducible example from the issue description.

**Comment by Malvoz** [comment]
@FloChehab Thanks for taking the time.

Seeing as some interactive elements such as `<button>` typically only receive the outline on focus and not on click I'm not opposed to this (e.g. the CSS solution in https://github.com/Leaflet/Leaflet/issues/8659#issuecomment-1610074905). I wonder why Chrome chose this behavior.

**Comment by pingumen96** [comment]
Thank Malvoz, sorry for creating a duplicate issue, I searched for it before but didn't found it.
I copy and paste my comment from the duplicated issue because maybe can help someone, that's how I solved the problem:

> I noticed that it happens due to the MultiPolygon, because if I bind the tooltip to a single feature inside the polygon it doesn't happen. I don't know if that's ok as default behaviour but putting the tooltip inside solved the problem for my use case

**Comment by natBizitza** [comment]
It's explained well [here](https://gis.stackexchange.com/questions/447502/prevent-leaflet-map-from-showing-rectangle-around-layer-on-click) and it's not an issue of this library as I understand.

**Comment by Falke-Design** [comment]
I think we should add the following rule so that the user can have the outline when tabbing, but is not disturbed when using the mouse.

```
.leaflet-container path.leaflet-interactive :focus:not(:focus-visible) {
  outline: 0;
}
```

Does anyone know any down sides of this?

**Comment by jonkoops** [comment]
That seems like a good solution to me @Falke-Design

**Comment by natBizitza** [comment]
> I think we should add the following rule so that the user can have the outline when tabbing, but is not disturbed when using the mouse.
> 
> ```
> .leaflet-container path.leaflet-interactive :focus:not(:focus-visible) {
>   outline: 0;
> }
> ```
> 
> Does anyone know any down sides of this?

This still shows the outline on mouse click in Chrome.

**Comment by jonkoops** [comment]
Perhaps we can do:

```css
:focus {
  outline: 0;
}

:focus-visible {
  outline: initial;
}
```

**Comment by natBizitza** [comment]
> Perhaps we can do:
> 
> ```css
> :focus {
>   outline: 0;
> }
> 
> :focus-visible {
>   outline: initial;
> }
> ```

This removes the visual outline completely for me.

**Comment by Falke-Design** [comment]
I just had a space to much in the css, this one is correct:

```
.leaflet-container path.leaflet-interactive:focus:not(:focus-visible) {
  outline: 0;
}
```

See: https://plnkr.co/edit/LaSeBR3E5z1Pak8d
